administrator/components/com_banners/src/Model/TracksModel.php

```diff
@@ -250,9 +250,12 @@ public function delete()
             }
         } else {
             Factory::getApplication()->enqueueMessage(Text::_('JERROR_CORE_DELETE_NOT_PERMITTED'), 'error');
```

Comment:
```suggestion
            $this->setError(Text::_('JERROR_CORE_DELETE_NOT_PERMITTED'), 'error');
```

otherwise the warning isn't going to show from the controller - but I guess instead the query from line 242 (which also looks wrong that it's set as the error?

Reply:
correct. please see new changes, the whole access check for banner tracks was wrong